src/lib/transaction_snark/test/zkapps_examples/initialize_state/initialize_state.ml

```diff
@@ -240,32 +241,42 @@ let%test_module "Initialize state test" =
           ; Update_state_party.party
           ]
       in
-      let zkapp_state = (Option.value_exn account.zkapp).app_state in
+      let zkapp_state =
+        (Option.value_exn (Option.value_exn account).zkapp).app_state
+      in
       Pickles_types.Vector.iter
         ~f:(fun x -> assert (Snark_params.Tick.Field.(equal one) x))
         zkapp_state
 
     let%test_unit "Update without initialize fails" =
       let account =
-        test_parties [ Deploy_party.party; Update_state_party.party ]
+        test_parties ~expected_failure:Account_precondition_unsatisfied
+          [ Deploy_party.party; Update_state_party.party ]
       in
-      assert (Option.is_none account.zkapp)
+      assert (Option.is_none (Option.value_exn account).zkapp)
 
     let%test_unit "Double initialize fails" =
       let account =
-        test_parties
+        test_parties ~expected_failure:Account_precondition_unsatisfied
           [ Deploy_party.party; Initialize_party.party; Initialize_party.party ]
       in
-      assert (Option.is_none account.zkapp)
+      assert (Option.is_none (Option.value_exn account).zkapp)
 
     let%test_unit "Initialize after update fails" =
       let account =
-        test_parties
+        test_parties ~expected_failure:Account_precondition_unsatisfied
           [ Deploy_party.party
           ; Initialize_party.party
           ; Update_state_party.party
           ; Initialize_party.party
           ]
       in
-      assert (Option.is_none account.zkapp)
+      assert (Option.is_none (Option.value_exn account).zkapp)
+
+    let%test_unit "Initialize without deploy fails" =
+      let account =
+        Option.try_with (fun () ->
+            test_parties [ Initialize_party.party; Update_state_party.party ])
+      in
+      assert (Option.is_none account)
```

Comment:
The fee payer and zkapp party are the same account so the account returned here should exist. I'm confused how this test is passing

Reply:
Applying the transaction fails a check in your harness. It's actually an `_ option option`, where the outer option falls due to the error.